docs/spark-writes.md

```diff
@@ -210,7 +210,7 @@ Branch writes can also be performed as part of a write-audit-publish (WAP) workf
 Note WAP branch and branch identifier cannot both be specified.
 Also, the branch must exist before performing the write. 
 The operation does **not** create the branch if it does not exist. 
-For more information on branches please refer to [branches](../tables/branching)
+For more information on branches please refer to [branches](../branching)
```

Comment:
Based on previous [review comment](https://github.com/apache/iceberg/pull/8165#pullrequestreview-1551505550), it should be `../tables/branching`

Reply:
+1

Reply:
@manuzhang @nastra thanks for looking at this and pointing the page to me. Just a nit, the current doc returns 404 not found when choosing **branches** in https://iceberg.apache.org/docs/1.3.1/spark-writes/#writing-to-branches section. The link shows https://iceberg.apache.org/docs/tables/branching but the correct link seems to be https://iceberg.apache.org/docs/<version>/branching. That's why I changed update the link.